cpp/velox/memory/VeloxColumnarBatch.cc

```diff
@@ -103,15 +72,35 @@ velox::RowVectorPtr VeloxColumnarBatch::getFlattenedRowVector() {
   return flattened_;
 }
 
-velox::RowVectorPtr convertBatch(
+std::shared_ptr<VeloxColumnarBatch> VeloxColumnarBatch::from(
     std::shared_ptr<facebook::velox::memory::MemoryPool> pool,
     std::shared_ptr<ColumnarBatch> cb) {
-  if (cb->getType() != "velox") {
-    auto vp = velox::importFromArrowAsOwner(*cb->exportArrowSchema(), *cb->exportArrowArray(), pool.get());
-    return std::dynamic_pointer_cast<velox::RowVector>(vp);
-  } else {
-    return std::dynamic_pointer_cast<VeloxColumnarBatch>(cb)->getRowVector();
+  if (cb->getType() == "velox") {
+    return std::dynamic_pointer_cast<VeloxColumnarBatch>(cb);
+  }
+  if (cb->getType() == "composite") {
+    auto composite = std::dynamic_pointer_cast<gluten::CompositeColumnarBatch>(cb);
```

Comment:
composite batch is only used in shuffle split range partition compute pid, after remove Arrow PR, this API will be deleted, we don't need to implement here

Reply:
No need to worry about that, we can either remove or it or not once we think it's no longer used. The code is independent enough to other parts.

Reply:
The new API 'addIntColumn' can be optimized out using `CompositeColumnarBatch` too. 

Composite can mean Velox+Velox as well as Velox+Arrow. After removing Arrow we can make PID column as a single column Velox batch.